toolchain/parse/node_kind.def

```diff
@@ -70,6 +70,30 @@ CARBON_PARSE_NODE_KIND_CHILD_COUNT(Name, 0,
 // A name in an expression context.
 CARBON_PARSE_NODE_KIND_CHILD_COUNT(NameExpression, 0, CARBON_TOKEN(Identifier))
 
+// ----------------------------------------------------------------------------
+
+// The comments below follow this pattern:
+
+// Descriptive heading:
+//   Child1
+//   Child2
+// Parent
+
+// In this, `Child1`, `Child2`, and `Parent` are all kinds of parse nodes, which
+// are then defined using the `CARBON_PARSE_NODE_KIND_*` macros. They are
+// written in the order they are emitted, and the indentation showing the tree
```

Comment:
```suggestion
// written in the order they are emitted, with the indentation showing the tree
```

"and the indentation shows" or "with the indentation showing" would both work, but I think "and the indentation showing" is grammatically incorrect.